tests/test_fastai_integration.py

```diff
@@ -0,0 +1,123 @@
+import os
+import shutil
+import time
+import uuid
+from unittest import TestCase, skip
+
+from huggingface_hub import HfApi
+from huggingface_hub.fastai_utils import (
+    from_pretrained_fastai,
+    push_to_hub_fastai,
+    save_fastai_learner,
+)
+from huggingface_hub.file_download import (
+    is_fastai_available,
+    is_fastcore_available,
+    is_torch_available,
+)
+
+from .testing_constants import ENDPOINT_STAGING, TOKEN, USER
+from .testing_utils import set_write_permission_and_retry
+
+
+def repo_name(id=uuid.uuid4().hex[:6]):
+    return "fastai-repo-{0}-{1}".format(id, int(time.time() * 10e3))
+
+
+WORKING_REPO_SUBDIR = f"fixtures/working_repo_{__name__.split('.')[-1]}"
+WORKING_REPO_DIR = os.path.join(
+    os.path.dirname(os.path.abspath(__file__)), WORKING_REPO_SUBDIR
+)
+
+if is_fastai_available():
+    from fastai.data.block import DataBlock
+    from fastai.test_utils import synth_learner
+
+if is_torch_available():
+    import torch
+
+
+def require_fastai_fastcore(test_case):
+    """
+    Decorator marking a test that requires fastai and fastcore.
+    These tests are skipped when fastai and fastcore are not installed.
+    """
+    if not is_fastai_available():
+        return skip("Test requires fastai")(test_case)
+    elif not is_fastcore_available():
+        return skip("Test requires fastcore")(test_case)
+    else:
+        return test_case
+
+
+def fake_dataloaders(a=2, b=3, bs=16, n=10):
+    def get_data(n):
+        x = torch.randn(bs * n, 1)
+        return torch.cat((x, a * x + b + 0.1 * torch.randn(bs * n, 1)), 1)
+
+    ds = get_data(n)
+    dblock = DataBlock()
+    return dblock.dataloaders(ds)
+
+
+if is_fastai_available():
+    DummyModel = synth_learner(data=fake_dataloaders())
```

Comment:
If this `DummyModel` is an instance of an object, can we please name without capital letters? 😄 

`dummy_model` would be fine imo